src/main/java/stirling/software/SPDF/controller/api/misc/ExtractImagesController.java

```diff
@@ -34,52 +16,78 @@
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 import org.springframework.web.multipart.MultipartFile;
-
-import io.github.pixee.security.Filenames;
-import io.swagger.v3.oas.annotations.Operation;
-import io.swagger.v3.oas.annotations.tags.Tag;
-
-import stirling.software.SPDF.model.api.PDFExtractImagesRequest;
-import stirling.software.SPDF.utils.ImageProcessingUtils;
+import stirling.software.SPDF.model.api.PDFWithImageFormatRequest;
 import stirling.software.SPDF.utils.WebResponseUtils;
+import stirling.software.SPDF.utils.memoryUtils;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.awt.image.RenderedImage;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
 
 @RestController
 @RequestMapping("/api/v1/misc")
-@Tag(name = "Misc", description = "Miscellaneous APIs")
 public class ExtractImagesController {
 
     private static final Logger logger = LoggerFactory.getLogger(ExtractImagesController.class);
+    private final Object lock = new Object();
+
+    private final memoryUtils memoryutils; // Inject MemoryUtils
+
+    public ExtractImagesController(memoryUtils memoryutils) {
+        this.memoryutils = memoryutils;
+    }
 
     @PostMapping(consumes = "multipart/form-data", value = "/extract-images")
     @Operation(
             summary = "Extract images from a PDF file",
             description =
                     "This endpoint extracts images from a given PDF file and returns them in a zip file. Users can specify the output image format. Input: PDF Output: IMAGE/ZIP Type: SIMO")
-    public ResponseEntity<byte[]> extractImages(@ModelAttribute PDFExtractImagesRequest request)
```

Comment:
You have removed isAllowDuplicates and using PDFWithImageFormatRequest instead of PDFExtractImagesRequest
Is this a bad merge?

Reply:
presume the other changes for this also missing

Reply:
Yes, updated now.